lib/core/include/scipp/core/element/map_to_bins.h

```diff
@@ -62,7 +62,7 @@ auto map_to_bins_chunkwise = [](auto &binned, auto &bins, const auto &data,
       const auto i_bin = bin_indices[i];
       if (i_bin < 0)
         continue;
-      const uint8_t j = i_bin % chunksize;
```

Comment:
Just for my understanding, why were we only seeing the bug for number of bins larger than > 65536 if the range of `uint8_t` is much smaller?
Also, why is `InnerIndex` `uint16_t` instead of `int16_t`?

Reply:
`chunksize` was set as `128` for less than `128**2` bins, `256` for less than 256**2` bins, etc.

Reply:
You mean `int16_t` instead of `uint16_t`? Since `int16_t` is large enough for our needs there is no reason for risky mixing of signed and unsigned integer arithmetic.

Reply:
ok thanks!